### PR TITLE
Payment issue 70

### PR DESCRIPTION
**Pull request by avonville**
@bwmarkle Code provided from mahesh prior to departure that fixed the transaction table issue. Also included emogrifier updates. Made sure that the code does not disrupt previous/pending merges with regards to weSerialize and PHP8 fixes (emogrifier regex update). 

To test broken state:
SCRIPT_DEBUG must be enabled to use the non-minified file.

From the master branch:

Create a payment form and proceed through the payment process.
image

** Note fix to frontend-form.js is added (weSerialize) or the form would not submit. **

- Payment information
email: elanad@inmotionhosting.com
card: 4242 4242 4242 4242
exp date: 04/24
cvc: 424
name: 424242424
zip: 90245

The transaction table id shows as zero instead of the entry id created in the entries table.

To Test Fix:

SCRIPT_DEBUG must be enabled to use the non-minified file.

From the Payment issue 70 branch:

Use the form created above and submit the entry and payment.

You will see that the transaction table id matches the entry id created

### CODE REVIEW

File: includes/class-ajax.php [hunk 1]
```diff
@@ -756,70 +753,64 @@ public function handle_frontend_submission() {
 
         $entry_fields = apply_filters( 'weforms_before_entry_submission', $entry_fields, $form, $form_settings, $form_fields );
 
-        $entry_id = weforms_insert_entry( [
-            'form_id' => $form_id,
-        ], $entry_fields );
-
+        $entry_id = 1;
```
Comment (bwmarkle) [inline includes/class-ajax.php:756]: To confirm, all this code is from Mahesh?

I'm not sure why we're setting `$entry_id = 1;` Isn't this the problem, the id is always 0 or 1?

**Review by bwmarkle — CHANGES_REQUESTED** [review]

File: assets/wpuf/js/frontend-form.js [hunk 2]
```diff
@@ -11,6 +11,20 @@
         });
     };
 
+    $.fn.extend({
+        /**
```
Comment (bwmarkle) [inline assets/wpuf/js/frontend-form.js:15]: Spacing of `/**`

Should look like this:
```js
/**
 * Short description. (use period)
 */
```

Comment (bwmarkle) [reply assets/wpuf/js/frontend-form.js:15]: Is this the method we're duplicating, both on the frontend and backend? If so, maybe a `@todo` comment mentioning this and we should fix it one day.

File: includes/class-ajax.php [hunk 3]
```diff
@@ -92,7 +92,7 @@ public function get_form() {
     public function save_form() {
         $post_data = wp_unslash( $_POST );
         if ( isset( $post_data['form_data'] ) ) {
-            parse_str( sanitize_text_field( wp_unslash( $post_data['form_data'] ) ),  $form_data );
+            parse_str( sanitize_text_field( wp_unslash( $post_data['form_data']  ) ),  $form_data );
```
Comment (bwmarkle) [inline includes/class-ajax.php:95]: No need for the extra space added.

File: includes/class-ajax.php [hunk 4]
```diff
@@ -756,70 +753,64 @@ public function handle_frontend_submission() {
...
-        } elseif ( $form_settings['redirect_to'] == 'url' ) {
-            $redirect_to = $form_settings['url'];
-        } elseif ( $form_settings['redirect_to'] == 'same' ) {
-            $show_message = true;
-        } else {
-            $show_message = true;
-        }
-
-        // Fire a hook for integration
+        //Fire a hook for integration
```
Comment (bwmarkle) [inline includes/class-ajax.php:768]: Why the change?

File: includes/class-ajax.php [hunk 5]
```diff
@@ -756,70 +753,64 @@ public function handle_frontend_submission() {
...
-            'data'         => $_POST,
-            'form_id'      => $form_id,
-            'entry_id'     => $entry_id,
-        ] );
-
         $notification = new WeForms_Notification( [
-            'form_id'  => $form_id,
-            'page_id'  => $page_id,
+            'form_id' => $form_id,
+            'page_id' => $page_id,
```
Comment (bwmarkle) [inline includes/class-ajax.php:772]: Why were the spaces removed?

**Review by bwmarkle — CHANGES_REQUESTED** [review]

File: assets/wpuf/js/frontend-form.js (hunk 2 above)
Comment (avonville) [reply assets/wpuf/js/frontend-form.js:15]: @bwmarkle  This should be removed and the `@todo` should be added to https://github.com/BoldGrid/weforms/pull/89

File: includes/class-ajax.php (hunk 5 above)
Comment (avonville) [reply includes/class-ajax.php:772]: Not sure, added them back so that they line up

File: includes/class-ajax.php (hunk 4 above)
Comment (avonville) [reply includes/class-ajax.php:768]: This was a mahesh change. But from what I see it was to make sure the entry is created/not created depending on the settings prior to any redirects after the form is submitted.

Comment (avonville) [reply includes/class-ajax.php]: If this talking about the space after `//` , this has been updated with the most recent commit.

**Review by bwmarkle — APPROVED** [review]
